go/tools/gazelle/rules/generator.go

```diff
@@ -146,22 +160,84 @@ func (g *generator) generateBin(rel, library string, pkg *build.Package) (*bzl.R
 	return newRule(kind, nil, attrs)
 }
 
-func (g *generator) generateLib(rel, name string, pkg *build.Package) (*bzl.Rule, error) {
+func (g *generator) generateLib(rel, name string, pkg *build.Package, cgoName string) (*bzl.Rule, error) {
 	kind := "go_library"
 
 	visibility := "//visibility:public"
 	// Libraries made for a go_binary should not be exposed to the public.
 	if pkg.IsCommand() {
 		visibility = "//visibility:private"
 	}
+
+	attrs := []keyvalue{
+		{key: "name", value: name},
+	}
+
+	if cgoName == "" {
+		srcs := append([]string{}, pkg.GoFiles...)
+		srcs = append(srcs, pkg.SFiles...)
+		attrs = append(attrs, keyvalue{key: "srcs", value: srcs})
+	} else {
+		// go_library gets mad when an empty slice is passed in, but handles not
+		// being set at all just fine when "library" is set.
+		if len(pkg.GoFiles) != 0 {
+			attrs = append(attrs, keyvalue{key: "srcs", value: pkg.GoFiles})
+		}
+		attrs = append(attrs, keyvalue{key: "library", value: ":" + cgoName})
+	}
 	visibility = checkInternalVisibility(rel, visibility)
 
+	attrs = append(attrs, keyvalue{key: "visibility", value: []string{visibility}})
+
+	deps, err := g.dependencies(pkg.Imports, rel)
+	if err != nil {
+		return nil, err
+	}
+	if len(deps) > 0 {
+		attrs = append(attrs, keyvalue{key: "deps", value: deps})
+	}
+
+	return newRule(kind, nil, attrs)
+}
+
+// Generates a cgo_library rule for C/C++ code.
```

Comment:
style: lead the comment with the func name
so
// generateCgoCLib generates a cgo_library rule for C/C++ code.